sui/src/sui_commands.rs

```diff
@@ -381,28 +322,13 @@ impl SuiNetwork {
 
 pub async fn genesis(
     genesis_conf: GenesisConfig,
-    single_address: Option<SuiAddress>,
 ) -> Result<(NetworkConfig, Vec<SuiAddress>, SuiKeystore), anyhow::Error> {
-    let num_to_provision = if single_address.is_none() {
-        genesis_conf.authorities.len()
-    } else {
-        1
-    };
+    let num_to_provision = genesis_conf.quorum_size;
 
     info!("Creating {} new authorities...", num_to_provision);
 
-    let mut network_config = NetworkConfig {
-        epoch: 0,
-        authorities: vec![],
-        buffer_size: 650000,
-        loaded_move_packages: vec![],
-        key_pair: genesis_conf.key_pair,
-    };
-    let mut voting_right = BTreeMap::new();
-    for authority in genesis_conf.authorities {
-        voting_right.insert(authority.public_key, authority.stake);
-        network_config.authorities.push(authority);
-    }
+    let config_dir = sui_config_dir().unwrap();
+    let mut network_config = NetworkConfig::generate(&config_dir, num_to_provision);
```

Comment:
repeating what brandon & I discussed offline:
> we’d like to keep the ability to genesis with a pre-specified validator sets (aka with their public keys). This is how it will work for testnet/mainnet and also help with debugability by keeping the same set of validator keys. People will run sui genesis with a genesis config and set up all the dbs etc etc. and we don’t need to socialize two files/config genesis.conf and network.conf

Reply:
in the end we should only need to provide validators with their own validator config.

If we want to fix the names of the validators for debug-ability there is another generate function which takes an RNG so we can just seed it